File: api/models/customer.py
from sqlalchemy import Column, String, Float
from models.base import TenantAwareModel

class Customer(TenantAwareModel):
    __tablename__ = "customers"

    first_name = Column(String(100), nullable=False)
    last_name = Column(String(100), nullable=False)
    biller = Column(String(255), nullable=True)
    business_number = Column(String(100), nullable=True)
    
    # Dual Addressing
    street_address_1 = Column(String(255))
    street_address_2 = Column(String(255))
    street_suburb = Column(String(100))
    street_city = Column(String(100))
    street_state = Column(String(100)) # Region
    street_country = Column(String(100))
    street_postcode = Column(String(20))
    
    postal_address_1 = Column(String(255))
    postal_address_2 = Column(String(255))
    postal_suburb = Column(String(100))
    postal_city = Column(String(100))
    postal_state = Column(String(100)) # Region
    postal_country = Column(String(100))
    postal_postcode = Column(String(20))
    
    # Contact Info
    phone = Column(String(50))
    mobile = Column(String(50))
    email = Column(String(255))
    fax = Column(String(50))
    preferred_contact_method = Column(String(50), default="EMAIL")
    
    # Financial Tiers
    hourly_rate = Column(Float, nullable=True) # Override workshop default
    discount_percent = Column(Float, default=0.0)
    markup_percent = Column(Float, default=0.0)
    payment_terms = Column(String(50), default="COD") # E.g. COD, NET30
    
    # Reference IDs
    imported_id = Column(String(100))
    government_id = Column(String(100))
    ams_member_number = Column(String(100))
    capricorn_member_number = Column(String(100))

File: web/src/types/customer.ts
export interface Customer {
    id: string;
    first_name: string;
    last_name: string;
    email: string | null;
    phone: string | null;
    mobile: string | null;
    fax: string | null;
    biller: string | null;

    // Status
    is_active: boolean;
    is_cash: boolean;
    is_individual: boolean;
    is_non_biller: boolean;

    // Financial
    unapplied_credit: number;
    account_balance: number;
    hourly_rate: number | null;
    discount: number | null;
    markup: number | null;

    // Addresses
    street_address_1: string | null;
    street_address_2: string | null;
    street_suburb: string | null;
    street_state: string | null;
    street_postcode: string | null;

    postal_address_1: string | null;
    postal_address_2: string | null;
    postal_suburb: string | null;
    postal_state: string | null;
    postal_postcode: string | null;

    // Contact Preferences
    web: string | null;
    preferred_contact_method: string | null;
}
